pampelmousse/exchange/api_handler.go

```
@@ -1,35 +1,53 @@
 package exchange
 
 import (
+	"fmt"
 	"math"
 
+	"errors"
+
 	"github.com/cs210/solvence-code/pampelmousse/util"
-	tspb "github.com/golang/protobuf/ptypes"
 	"github.com/jinzhu/gorm"
 	context "golang.org/x/net/context"
 )
 
 // Server represents the gRPC server
 type Server struct {
-	Database *gorm.DB
+	DB *gorm.DB
+}
+
+//NewServer creates a new Server
+func NewServer(db *gorm.DB) *Server {
+	s := new(Server)
+	s.DB = db
+	return s
 }
 
 //CreateLimitBuyOrder creates a sell order specified by the Order struct
 func (s *Server) CreateLimitBuyOrder(ctx context.Context, in *Order) (*Empty, error) {
-	createLimitOrder(in.GetPrice(), in.GetQuantity(), int(in.GetUserID()), int(in.GetCoinID()), s.Database, util.LimitBuy)
+	if in == nil {
+		return nil, errors.New("Order request param 'in' is null")
+	}
+	createLimitOrder(in.GetPrice(), in.GetQuantity(), int(in.GetUserID()), int(in.GetCoinID()), s.DB, util.LimitBuy)
```

A returned OrderResponse object would be great here to confirm with the frontend, unless that is happening somewhere else?

---

@geoffreyangus its an async operation so itll be on the books but may or may not be complete. I think it might be easier for us to just query for all users open orders right after creating one to show progress or show its completed.

---

Ooooh gotcha. Do you think we should be displaying the user a "Order received" success response message through the backend? I think some guarantee on the frontend that the user is all good would be great. I can just do some check mark animation on my end with no backend guarantee as well if a response seems extraneous

---

@geoffreyangus if there is a problem I will return an error rather than Empty, so getting an Empty response should signal its all good.